packages/playwright-core/src/client/locator.ts

```diff
@@ -31,10 +31,47 @@ export type LocatorOptions = {
   has?: Locator;
 };
 
+export type ByRoleOptions = LocatorOptions & {
+  checked?: true | false | 'mixed';
+  disabled?: boolean;
+  expanded?: boolean;
+  includeHidden?: boolean;
+  level?: number;
+  name?: string | RegExp;
+  pressed?: true | false | 'mixed';
+};
+
 export class Locator implements api.Locator {
   _frame: Frame;
   _selector: string;
 
+  static getByTextSelector(text: string | RegExp, options?: { exact?: boolean }): string {
+    if (!isString(text))
+      return `text=${text}`;
+    const escaped = JSON.stringify(text);
+    const selector = options?.exact ? `text=${escaped}` : `text=${escaped.substring(1, escaped.length - 1)}`;
```

Comment:
I am not exactly sure that non-quoted escaped text is what we need. We should double check whether `text=foo\"bar` actually parses with slash or without it.